src/domain/discounts/discount-form/condition-tables/selectable-table.tsx

```diff
@@ -1,201 +1,175 @@
+import { debounce } from "lodash"
+import React, { useEffect } from "react"
 import {
-  CustomerGroup,
-  Product,
-  ProductCollection,
-  ProductTag,
-  ProductType,
-} from "@medusajs/medusa"
-import React, { useEffect, useMemo } from "react"
-import {
-  ColumnInstance,
+  Column,
+  HeaderGroup,
+  Row,
   usePagination,
   useRowSelect,
   useTable,
 } from "react-table"
 import Spinner from "../../../../components/atoms/spinner"
 import IndeterminateCheckbox from "../../../../components/molecules/indeterminate-checkbox"
-import Table, { TablePagination } from "../../../../components/molecules/table"
-import { PaginationProps } from "../../../../types/shared"
+import Table, {
+  TablePagination,
+  TableProps,
+} from "../../../../components/molecules/table"
+import useQueryFilters from "../../../../hooks/use-query-filters"
 
-type SelectableTableProps = {
-  showSearch?: boolean
-  objectName?: string
+type SelectableTableProps<T extends object> = {
+  resourceName?: string
   label?: string
   isLoading?: boolean
-  pagination: PaginationProps
-  totalCount?: number
-  data?:
-    | Product[]
-    | ProductType[]
-    | ProductCollection[]
-    | ProductTag[]
-    | CustomerGroup[]
+  totalCount: number
+  options: Omit<TableProps, "filteringOptions"> & {
+    filters?: Pick<TableProps, "filteringOptions">
+  }
+  data?: T[]
   selectedIds?: string[]
-  columns: Partial<ColumnInstance>[]
-  onPaginationChange: (pagination: PaginationProps) => void
+  columns: Column<T>[]
   onChange: (items: string[]) => void
-  onSearch?: (search: string) => void
-}
+  renderRow: (props: { row: Row<T> }) => React.ReactElement
+  renderHeaderGroup?: (props: {
+    headerGroup: HeaderGroup<T>
+  }) => React.ReactElement
+} & ReturnType<typeof useQueryFilters>
 
-export const SelectableTable: React.FC<SelectableTableProps> = ({
-  showSearch = true,
+export const SelectableTable = <T extends object>({
```

Comment:
I think we could be a bit more precise here and narrow the type: `T = Product | CustomerGroup | ...`

Reply:
I think we might be extracting this table down the line so that we can use it across different domains including, but not limited to the ones you suggested. Nevertheless, I'll make the change. Perhaps it's better to narrow right now given that we know for a fact that we'll be using it across those domains 😄 . Lmk what you think 👍

Reply:
It makes sense to me, let's leave it this way now since you already made the change and when we will extract the table to be a template we can change the type.

Reply:
Can we merge this? @fPolic @zakariaelas